backend/src/packages/business/business.controller.ts

```diff
@@ -831,6 +865,87 @@ class BusinessController extends Controller {
       payload: deletionResult,
     };
   }
+
+  /**
+   * @swagger
+   * /business/trucks:
+   *    get:
+   *      tags:
+   *       - business/trucks
+   *      summary: Find all trucks
+   *      description: Find all trucks
+   *      security:
+   *        - bearerAuth: []
+   *      responses:
+   *        200:
+   *          description: Successful find all trucks
+   *          content:
+   *            application/json:
+   *              schema:
+   *                type: array
+   *                items:
+   *                  $ref: '#/components/schemas/TruckResponse'
+   */
+
+  private async findAllTrucks(
+    options: ApiHandlerOptions<{
+      query: PaginationPayload;
+      user: UserEntityObjectWithGroupT;
+    }>,
+  ): Promise<ApiHandlerResponse> {
+    const trucks = await this.businessService.findAllTrucksByBusinessId(
+      options.user.id,
```

Comment:
it's user id from users table, it's not business id
in services you should find business id by authorized user id

Reply:
This is the id of an authorized user who is a business. And in the service, I find the businessId like this: 

`   const business = await this.findByOwnerId(businessId);

    if (!business) {
      throw new HttpError({
        status: HttpCode.BAD_REQUEST,
        message: HttpMessage.BUSINESS_DOES_NOT_EXIST,
      });
    }

    return await this.truckService.findAllByBusinessId(business.id, query); `

The only thing I can do is to name the method in the service differently to make it clearer that I am passing the id of the authorized user to the service

Reply:
yeap, left comment below